### PR TITLE
Updated readme and docs to point at dev.tillitis.se.

### PR DESCRIPTION
**Pull request by dehanj**
Updated readme and docs that has been outdated. 
Kept old documentation for the sake of history, but added a note pointing towards dev.tillitis.

### CODE REVIEW

File: doc/system_description/boards.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Tillitis PCB designs
 
+**NOTE:** Documentation migrated to dev.tillitis.se, this is kept for
```
Comment (mchack-work) [inline doc/system_description/boards.md:3]: The boards document isn't in Dev Handbook yet. Maybe it should stay here and not have any warning?

File: doc/system_description/fpga.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # The TKey FPGA
 
+**NOTE:** Documentation migrated to dev.tillitis.se, this is kept for
```
Comment (mchack-work) [inline doc/system_description/fpga.md:3]: The FPGA document hasn't been moved to the Dev Handbook either. Maybe it should stay?

**Review by mchack-work — COMMENTED** [review]
Very good to change the README and the links. I think adding the warning to the documents that have moved to Dev Handbook is also good, but perhaps create an issue to actually delete them later?

I've added comments about some of the documents whose contents *haven't* moved to the Dev Handbook. They should probably stay close to the source? It's about the PCB and the FPGA internals. Might want to move them up from system_description, perhaps... But I think they should stay in the repo.

File: doc/system_description/boards.md (hunk 1 above)
Comment (dehanj) [reply doc/system_description/boards.md:3]: Good catch. Will remove the note.

File: doc/system_description/fpga.md (hunk 2 above)
Comment (dehanj) [reply doc/system_description/fpga.md:3]: Good catch. Will remove the note.

**Review by mchack-work — APPROVED** [review]
LGTM.

**Pull request merged**